src/lib/server/generateStaticHTMLFile.js

```diff
@@ -0,0 +1,33 @@
+import path from "path";
+import { prepareOutput } from "./RequestHandler";
+import { getLogger } from "./logger";
+import WebpackIsomorphicTools from "webpack-isomorphic-tools";
+import webpackConfig from "../../config/webpack-isomorphic-tools-config";
+import detectEnvironmentVariables from "../detectEnvironmentVariables";
```

Comment:
`detectEnvironmentVariables` is one of my favorite GlueStick features. When you build an app with GlueStick, the src/config/application.js file sometimes references environment variables for setting different config options.  Environment variables in node are accessed using `process.env.______`. Since this code is going to be run in the browser we needed a way to pass that data long to the browser. However, we don't want to pass all environment variables to the browser or else you would be leaking all kinds of stuff like AWS keys and other environment variables that need to be private. That is why you see a warning at the top of src/config/application.js that says any environment variables used in src/config/application.js will exposed in the browser.

We also didn't want to hard code the environment variables into the build because if you run the same build in multiple pods it is pretty common to change environment variables for different pod environments. So to deal with all of this we convert src/config/application.js into an AST and process it to find any references to an environment variable. Then we replace those environment variables with `window.__GS_ENV_VARS.____` and recompile the code. We also inject the actual values for those environment variables into the page.

Anyways, none of that needs to be updated but I figured it helps to understand what it is used for.